piper/ratbagd.py

```diff
@@ -206,14 +206,15 @@ class Ratbagd(_RatbagdDBus):
     def __init__(self):
         _RatbagdDBus.__init__(self, "Manager", None)
 
+        # FIXME: if we start adding and removing objects from this list,
+        # things will break!
+        result = self._get_dbus_property("Devices")
+        self._devices = [RatbagdDevice(objpath) for objpath in result]
+
     @GObject.Property
     def devices(self):
         """A list of RatbagdDevice objects supported by ratbagd."""
-        devices = []
-        result = self._get_dbus_property("Devices")
-        if result is not None:
-            devices = [RatbagdDevice(objpath) for objpath in result]
-        return devices
+        return self._devices
```

Comment:
hmm, I'm fine with the rest of the changes, but this one we need to really fix. (un)plugging a device while ratbagd is running *must* work.